[GUIDE] Query KYC State on C-Chain with Remix IDE (#90)
chain4travel/camino-docs: docs/guides/query-kyc-state-on-c-chain.md

---
sidebar_position: 45
title: Query KYC State on C-Chain
description: This document describes how to use Remix IDE to query KYC State of an address on C-Chain
---

# How to Query KYC State on C-Chain

> This document provides a guide on how to use Remix IDE to query the KYC state of

Reviewer: might be useful to add the link to remix: https://remix.ethereum.org/

Author: We already have the link down below where it's needed. But I will add it into the this part as well.

an address on C-Chain. Follow the steps below to accomplish this.

> ## Requirements

Reviewer: Might be useful to add to the requirements:
"
- Remix IDE
- CaminoETHVM ABI file  
"

Author: These are not prerequisites, the document handles these. Maybe I should change the title to Prerequisites instead of Requirements?

- MetaMask installation and functionality

## Add Camino Network to MetaMask

To query the KYC state of an address using the Remix IDE, we will use the MetaMask wallet.

> We need to add a custom network to MetaMask. In this document, we will use the Columbus test network.

Reviewer: just to be sure this is was done with the new Columbus network, correct? 
Some partners have been using the old test network, so might be helpful to specify.

Author: This is hard to specify and the old testnet will be gone. So let's keep it like this.

<figcaption align="center"><b>Fig.3:</b> Fill out the text fields</figcaption>
</figure>

- Verify that the custom network has been added successfully.

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/3-add-custom-network-success.png"/>
<figcaption align="center"><b>Fig.4:</b> Verify success</figcaption>
</figure>

## Connecting MetaMask to the Remix IDE

Follow the steps below to connect MetaMask to the IDE:

- Open the Remix IDE web page: https://remix.ethereum.org/
- Click **Deploy and run transaction** on the right sidebar.
- Select **Inject Provider - MetaMask** from the **Environment** dropdown menu.

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/4-select-injected-provider.png"/>
<figcaption align="center"><b>Fig.5:</b> Click "Deploy and run transactions" tab and select "Inject Provider" from "Environment"</figcaption>
</figure>

- Verify that the MetaMask is connected to the Remix IDE.

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/5-check-network-connected.png"/>
<figcaption align="center"><b>Fig.6:</b> Verify network is connected</figcaption>
</figure>

## Downloading the CaminoAdmin.abi file

> We can either compile the contract or use the precompiled ABI file. In this guide, we will use the ABI file.

Reviewer: @jax what would be the main difference process-wise if the developer was to not use the precompiled ABI file?

<figcaption align="center"><b>Fig.9:</b> Select the loaded ABI file</figcaption>
</figure>

## Getting the CaminoAdmin Contract Address

**Camino Admin Address**: `0x010000000000000000000000000000000000000a`

You can also find the Camino contract addresses on [Camino Smart Contracts](/developer/references/camino-contracts.md) page under `Developer` > `References`.

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/9-get-the-contract-address.png"/>
<figcaption align="center"><b>Fig.10:</b> Get the CaminoAdmin contract address</figcaption>
</figure>

## Interacting with the Address

- Paste the address into the **At Address** field and click on the button.

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/10-paste-address-and-click-at-address.png"/>
<figcaption align="center"><b>Fig.11:</b> Paste the contract address into the "At Address" field and click "At Address"</figcaption>
</figure>

- Confirm the interaction

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/11-confirm-the-popup.png"/>
<figcaption align="center"><b>Fig.12:</b> Confirm the dialog popup</figcaption>
</figure>

## Making the Query

> Follow the steps below to make the query:

Reviewer: This is the manual query of course, it might be useful to specify how this can be written into the smart contract, without necessarily giving the step-by-step, but at least being very clear that the check can be automated.

- Click the arrow to open the contracts methods list.

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/12-open-dropdown-abi.png"/>
<figcaption align="center"><b>Fig.13:</b> Click the arrow to open the dropdown</figcaption>
</figure>

> - Paste the address you want to query KYC state into the field and click **`getKycState`** button.

Reviewer: I think it might be important here to specify that it must be an address that starts with "0x" and needs to be the c-chain address, as it cannot even be the hashed public address that resolves all subnets I believe.

Author: Of course, this is a documentation for the C-Chain. Other chains have native API methods to query an address state (KYC etc). I'll add a note about this.

:::note

This is the C-Chain address of a wallet, that starts with `0x`.

If you want to query the **P-Chain address**,
please refer to the [`platform.getAddressStates()`](/developer/apis/camino-node-apis/p-chain#platformgetaddressstates)
method in P-Chain API documentation.

:::

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/13-paste-the-address-n-click-getkycstate.png"/>
<figcaption align="center"><b>Fig.14:</b> Paste the address to query and click getKycState button</figcaption>
</figure>

## Result

- The result of the query will be displayed below the **`getKycState`** button.

### Example Result for KYC Verified Address

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/14-kyc-state-result.png"/>
<figcaption align="center"><b>Fig.15:</b> Example result for KYC Verified address</figcaption>
</figure>

### Example Result for Non-Verified Address

<figure>
<img class="zoom" src="/img/query-kyc-state-c-chain/15-kyc-state-result-zero.png"/>
<figcaption align="center"><b>Fig.15:</b> Example result for Non-Verified address</figcaption>
</figure>

:::info Conclusion

> Smart contracts on the C-Chain can utilize the CaminoAdmin contract to inquire and authenticate

Reviewer: ```suggestion
Smart contracts on the C-Chain request information from the CaminoAdmin contract to inquire and authenticate
```
